### PR TITLE
Portworx module

### PR DESCRIPTION
**Pull request by cambelem**
Created a portworx module that will provision block storage volumes, attach, and finally setup portworx.

### CODE REVIEW

File: portworx/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+## Set up access to IBM Cloud
+
+If running these modules from your local terminal, you need to set the credentials to access IBM Cloud.
+
+Go [here](../CREDENTIALS.md) for details.
+
+## Provisioning this module in a Terraform Script
+
+In your Terraform code define the `ibm` provisioner block with the `region` and the `generation`, which is **1 for Classic** and **2 for VPC Gen 2**.
```
Comment (annumberhocker) [inline portworx/README.md:24]: `generation` isn't needed anymore.

File: portworx/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+### Provisioning the Portworx Module
+
+Use a `module` block assigning `source` to `git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//portworx`. Then set the [input variables](#input-variables) required to install the Portworx service.
+
+```hcl
+module "portworx" {
+  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//portworx"
+  enable = true
+  ibmcloud_api_key = "api-key"
```
Comment (annumberhocker) [inline portworx/README.md:46]: should this be listed as ibmcloud_api_key = "<api-key>". 

This shouldn't literally be "api-key", right?

File: portworx/testing/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+### Using Terraform client
+
+Follow these instructions to test the Terraform Module manually
+
+Create the file `test.auto.tfvars` with the following input variables, these values are fake examples:
+
+```hcl
+enable                  = true
+ibmcloud_api_key        = "api-key"
```
Comment (annumberhocker) [inline portworx/testing/README.md:23]: same as above.  This should indicated "<api-key>"

File: portworx/testing/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+// Storage parameters
+install_storage         = true
+storage_capacity        = 200  // In GBs
+storage_iops            = 10   // Must be a number, it will not be used unless a storage_profile is set to a custom profile
+storage_profile         = "10iops-tier"
+
+// Portworx parameters
+resource_group_name     = "default"
+region                  = "us-east"
+cluster_id              = "cluster-id"
```
Comment (annumberhocker) [inline portworx/testing/README.md:38]: "<cluster_id>"

File: portworx/portworx.tf [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  location          = var.region
+  resource_group_id = data.ibm_resource_group.group.id
+
+  tags = [
+    "clusterid:${var.cluster_id}",
+  ]
+
+  parameters = {
+    apikey           = var.ibmcloud_api_key
+    cluster_name     = "pwx"
```
Comment (annumberhocker) [inline portworx/portworx.tf:156]: Is this intentially hardcoded here?

File: portworx/portworx.tf [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  plan              = "px-enterprise"
+  location          = var.region
+  resource_group_id = data.ibm_resource_group.group.id
+
+  tags = [
+    "clusterid:${var.cluster_id}",
+  ]
+
+  parameters = {
+    apikey           = var.ibmcloud_api_key
```
Comment (annumberhocker) [inline portworx/portworx.tf:155]: Where is this actually used?

**Review by annumberhocker — COMMENTED** [review]
See my comments

File: portworx/portworx.tf (hunk 6 above)
Comment (cambelem) [reply portworx/portworx.tf:155]: It's needed to provision the Portworx service.

File: portworx/portworx.tf (hunk 5 above)
Comment (cambelem) [reply portworx/portworx.tf:156]: This is intentional. This cluser_name is not the same as the user's cluster name.

Portworx cluster name:
Enter a unique name for the storage layer that Portworx creates within your Kubernetes or OpenShift cluster.

**Pull request merged**